packages/hdwallet-ledger/src/bitcoin.ts

```diff
@@ -53,45 +56,27 @@ export async function btcGetPublicKeys(
   const xpubs: Array<core.PublicKey | null> = [];
 
   for (const getPublicKey of msg) {
-    let { scriptType } = getPublicKey;
     const { addressNList, coin } = getPublicKey;
 
     if (!coin) throw new Error("coin is required");
+    if (!getPublicKey.scriptType) throw new Error("scriptType is required");
 
-    const parentBip32path: string = core.addressNListToBIP32(addressNList.slice(0, -1)).substring(2); // i.e. "44'/0'"
-    const bip32path: string = core.addressNListToBIP32(addressNList).substring(2); // i.e 44'/0'/0'
+    const parentBip32path: string = core.addressNListToBIP32(addressNList);
 
-    const opts = { verify: false };
-
-    const res1 = await transport.call("Btc", "getWalletPublicKey", parentBip32path, opts);
-    handleError(res1, transport, "Unable to obtain public key from device.");
-
-    const {
-      payload: { publicKey: parentPublicKeyHex },
-    } = res1;
-    const parentPublicKey = compressPublicKey(Buffer.from(parentPublicKeyHex, "hex"));
-    const parentFingerprint = new DataView(
-      bitcoin.crypto.ripemd160(bitcoin.crypto.sha256(parentPublicKey)).buffer
-    ).getUint32(0);
-
-    const res2 = await transport.call("Btc", "getWalletPublicKey", bip32path, opts);
-    handleError(res2, transport, "Unable to obtain public key from device.");
-
-    const {
-      payload: { publicKey: publicKeyHex, chainCode: chainCodeHex },
-    } = res2;
-    const publicKey = compressPublicKey(Buffer.from(publicKeyHex, "hex"));
-    const chainCode = Buffer.from(chainCodeHex, "hex");
+    const getWalletXpubResponse = await transport.call("Btc", "getWalletXpub", {
```

Comment:
New method of `@ledgerhq/hw-app-btc` for getting xpubs without the previous cognitive gymnastics of the previous implementation